### ISSUE TITLE
Enhancement for Analytics Tracking for Email Share functionality

### ISSUE THREAD

**Issue by skumuda** [adobe/asset-share-commons#1208]
Hi,

Can we add analytics tracking for share functionality in ASC?
Since analytics we use on all websites to track the user-data. I would request if we can add a query parameter at the end of url that gets shared over the email.

Solution:
Can we create a new tab 'Analytics' in E-mail Share modal dialog which is under /content/asset-share-commons/en/light/actions/share.html. Under Analytics tab create a new field 'Query Param'. This value will be picked and appended in assetLinkHTML for every link.

assetLinkHTML is getting generated in EmailShareServiceImpl in getAssetLinkListHtml() line number 168 where url is appended. Here we can add the query param value that comes from dialog.

It would be helpful if this can be achieved and provided as a part of ServicePack.

**Comment by davidjgonzalez** [comment]
Hi @skumuda - if we set it at the Dialog level, this will set the same value for every share link -- is that what youre going for? You just want to know what Details page views are from shares vs navigating via search/bookmarks?

**Comment by skumuda** [comment]
@davidjgonzalez Yes, same value should be passed for every share link. This is to track which assets are shared.

**Comment by skumuda** [comment]
Hi @davidjgonzalez Please let me know your thoughts/suggestion on the above implementation.

**Comment by davidjgonzalez** [comment]
Released in v3.9.0 - If you goto the share modal's dialog, you can put in tracking a query param to add
